### ISSUE TITLE
Device not recognized

### ISSUE THREAD

**Issue by risinek** [openstf/stf#930]
Yesterday I updated Android SDK on Mac Mini to 28. Since then STF won't recognise devices connected to the machine while stf is already running. It recognise devices connected to Mac before stf is started.

stf doctor output: 
```
2018-08-10T09:37:35.033Z INF/cli:doctor 30132 [*] OS Arch: x64
2018-08-10T09:37:35.035Z INF/cli:doctor 30132 [*] OS Platform: darwin
2018-08-10T09:37:35.035Z INF/cli:doctor 30132 [*] OS Platform: 17.7.0
2018-08-10T09:37:35.035Z INF/cli:doctor 30132 [*] Using Node 8.0.0
2018-08-10T09:37:35.054Z INF/cli:doctor 30132 [*] Using ZeroMQ 4.2.2
2018-08-10T09:37:35.080Z INF/cli:doctor 30132 [*] Using GraphicsMagick 1.3.25
2018-08-10T09:37:35.081Z INF/cli:doctor 30132 [*] Using ProtoBuf 3.3.0
2018-08-10T09:37:35.081Z ERR/cli:doctor 30132 [*] ADB version cannot be detected
2018-08-10T09:37:35.083Z INF/cli:doctor 30132 [*] Using RethinkDB 2.3.5
```

**Comment by sorccu** [comment]
Since it says `ADB version cannot be detected`, it sounds to me like there's something wrong with your SDK/ADB installation. I don't think we can help with this, but please update the issue if you find the cause.

**Comment by risinek** [comment]
Well, `adb` works fine on that machine. I installed SDK using brew, and platform-tools using sdkmanager from SDK. Where from STF is getting ADB path? 
Weird part is, that STF is still usable for devices, that are connected to it from the start. So STF has to have access to ADB..

**Comment by risinek** [comment]
I just checked logs and found this:
```
2018-08-10T09:47:30.988Z INF/provider 30227 [*] Found device "52037268eeb183f7" (connecting)
Unhandled rejection Error: Illegal value for Message.Field .DeviceIntroductionMessage.status of type enum: undefined (not a valid enum value)
    at Field.<anonymous> (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:2641:27)
    at Field.ProtoBuf.Reflect.FieldPrototype.verifyValue (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:2738:25)
    at MessagePrototype.set (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:1799:63)
    at Message (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:1728:42)
    at /Users/brnorob/stf-modified/lib/units/provider/index.js:165:29
    at /Users/brnorob/stf-modified/lib/units/provider/index.js:161:22
    at Tracker.<anonymous> (/Users/brnorob/stf-modified/lib/units/provider/index.js:144:16)
    at emitOne (events.js:115:13)
    at Tracker.emit (events.js:210:7)
    at Tracker.update (/Users/brnorob/stf-modified/node_modules/adbkit/lib/adb/tracker.js:62:14)
    at /Users/brnorob/stf-modified/node_modules/adbkit/lib/adb/tracker.js:38:15
    at runCallback (timers.js:800:20)
Unhandled rejection Error: Illegal value for Message.Field .DeviceIntroductionMessage.status of type enum: undefined (not a valid enum value)
    at Field.<anonymous> (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:2641:27)
    at Field.ProtoBuf.Reflect.FieldPrototype.verifyValue (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:2738:25)
    at MessagePrototype.set (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:1799:63)
    at Message (/Users/brnorob/stf-modified/node_modules/protobufjs/dist/ProtoBuf.js:1728:42)
    at /Users/brnorob/stf-modified/lib/units/provider/index.js:165:29
    at /Users/brnorob/stf-modified/lib/units/provider/index.js:161:22
    at Tracker.<anonymous> (/Users/brnorob/stf-modified/lib/units/provider/index.js:144:16)
    at emitOne (events.js:115:13)
    at Tracker.emit (events.js:210:7)
    at Tracker.update (/Users/brnorob/stf-modified/node_modules/adbkit/lib/adb/tracker.js:62:14)
    at /Users/brnorob/stf-modified/node_modules/adbkit/lib/adb/tracker.js:38:15
    at runCallback (timers.js:800:20)
2018-08-10T09:47:41.007Z INF/provider 30227 [*] Providing all 3 of 5 device(s); ignoring "42009af4fab8b300", "52037268eeb183f7"
```

**Comment by sorccu** [comment]
You could have searched past issues. You're hitting https://github.com/openstf/stf/pull/913. It was fixed in 3.3.1 as you can see in the [CHANGELOG](https://github.com/openstf/stf/blob/master/CHANGELOG.md#331-2018-07-14).

**Comment by risinek** [comment]
Sorry I didn't checked logs before.. Thanks!
